cdap-security/src/main/java/io/cdap/cdap/security/tools/AccessTokenGeneratorService.java

```diff
@@ -0,0 +1,228 @@
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import io.cdap.cdap.api.metrics.MetricsCollectionService;
+import io.cdap.cdap.common.BadRequestException;
+import io.cdap.cdap.common.HttpExceptionHandler;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.common.conf.SConfiguration;
+import io.cdap.cdap.common.guice.ConfigModule;
+import io.cdap.cdap.common.guice.IOModule;
+import io.cdap.cdap.common.http.CommonNettyHttpServiceFactory;
+import io.cdap.cdap.common.io.Codec;
+import io.cdap.cdap.common.metrics.NoOpMetricsCollectionService;
+import io.cdap.cdap.common.security.HttpsEnabler;
+import io.cdap.cdap.security.auth.AccessToken;
+import io.cdap.cdap.security.auth.TokenManager;
+import io.cdap.cdap.security.auth.UserIdentity;
+import io.cdap.cdap.security.guice.CoreSecurityRuntimeModule;
+import io.cdap.http.AbstractHttpHandler;
+import io.cdap.http.HttpResponder;
+import io.cdap.http.NettyHttpService;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import org.apache.commons.cli.BasicParser;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.Base64;
+import java.util.Collections;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
+
+/**
+ * A service for managing the lifecycle of an http service that serves access tokens.
+ */
+public class AccessTokenGeneratorService extends AbstractIdleService {
+  // Http handler configuration
+  static final String ADDRESS_CONFIG = "access.token.generator.bind.address";
+  static final String ADDRESS_DEFAULT = "127.0.0.1";
+  static final String PORT_CONFIG = "access.token.generator.bind.port";
+  static final int PORT_DEFAULT = 11030;
+  // HttpHandler has only one caller, and is accessed infrequently so one worker thread is sufficient
+  static final String WORKER_POOL_SIZE_CONFIG = "access.token.generator.worker.pool.size";
+  static final int WORKER_POOL_SIZE_DEFAULT = 1;
+  static final String EXEC_POOL_SIZE_CONFIG = "access.token.generator.executor.pool.size";
+  static final int EXEC_POOL_SIZE_DEFAULT = 0;
+
+  private final NettyHttpService httpService;
+  private final TokenGeneratorHttpHandler handler;
+
+  @Inject
+  AccessTokenGeneratorService(CConfiguration cConf, SConfiguration sConf,
+                              CommonNettyHttpServiceFactory httpServiceFactory,
+                              TokenManager tokenManager,
+                              Codec<AccessToken> tokenCodec) {
+    handler = new TokenGeneratorHttpHandler(cConf, tokenManager, tokenCodec);
```

Comment:
Seems like you can just have the `TokenGeneratorHttpHandler` injected through the constructor directly.

Reply:
fixed